Panic during TestTaintNodeByCondition integration test

[Panic](https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/pr-logs/pull/61976/pull-kubernetes-integration/9691/) in the `TestTaintNodeByCondition` integration test in an seemingly unrelated PR https://github.com/kubernetes/kubernetes/pull/61976. 

```
panic: runtime error: invalid memory address or nil pointer dereference
/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:58 +0x107
/usr/local/go/src/runtime/panic.go:502 +0x229
/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/apis/meta/v1/meta.go:133
/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/scheduler/core/equivalence_cache.go:89 +0xc4
/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/scheduler/core/generic_scheduler.go:512 +0x1de
/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/scheduler/core/generic_scheduler.go:353 +0x19a
/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/client-go/util/workqueue/parallelizer.go:47 +0x96
/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/client-go/util/workqueue/parallelizer.go:43 +0x10b
```

@kubernetes/sig-scheduling-bugs

still seeing this:

panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x1103224]

goroutine 47096 [running]:
k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:58 +0x107
panic(0x361b8e0, 0x71ac100)
	/usr/local/go/src/runtime/panic.go:502 +0x229
k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/apis/meta/v1.(*ObjectMeta).GetName(...)
	/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/apis/meta/v1/meta.go:133
k8s.io/kubernetes/pkg/scheduler/core.(*EquivalenceCache).RunPredicate(0xc42500cd40, 0x3f62748, 0x3e2aa3f, 0x11, 0xc42f9a2380, 0x523ab40, 0xc4345c8e10, 0x0, 0xc4235e5d50, 0x528e060, ...)
	/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/scheduler/core/equivalence_cache.go:76 +0xc4
k8s.io/kubernetes/pkg/scheduler/core.podFitsOnNode(0xc42f9a2380, 0x523ab40, 0xc4345c8e10, 0x0, 0xc4339a3320, 0x528e060, 0xc42436fa40, 0xc42500cd40, 0x527b2c0, 0xc43b40ee70, ...)
	/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/scheduler/core/generic_scheduler.go:512 +0x1de
k8s.io/kubernetes/pkg/scheduler/core.(*genericScheduler).findNodesThatFit.func1(0x0)
	/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/scheduler/core/generic_scheduler.go:353 +0x19a
k8s.io/kubernetes/vendor/k8s.io/client-go/util/workqueue.Parallelize.func1(0xc4235e5d90, 0xc42cdbad20, 0xc43bb83dc0)
	/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/client-go/util/workqueue/parallelizer.go:47 +0x96
created by k8s.io/kubernetes/vendor/k8s.io/client-go/util/workqueue.Parallelize
	/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/client-go/util/workqueue/parallelizer.go:43 +0x10b

https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/pr-logs/pull/64028/pull-kubernetes-integration/12124/

--- a/pkg/scheduler/core/equivalence_cache.go
+++ b/pkg/scheduler/core/equivalence_cache.go
@@ -17,6 +17,7 @@ limitations under the License.
 package core
 
 import (
+	"fmt"
 	"hash/fnv"
 	"sync"
 
@@ -86,6 +87,12 @@ func (ec *EquivalenceCache) RunPredicate(
 ) (bool, []algorithm.PredicateFailureReason, error) {
 	ec.mu.Lock()
 	defer ec.mu.Unlock()
+
+	if nodeInfo == nil || nodeInfo.Node() == nil {
+		// This may happen during tests.
+		return false, []algorithm.PredicateFailureReason{}, fmt.Errorf("nodeInfo is nil or node is invalid")
+	}
+
 	fit, reasons, invalid := ec.lookupResult(pod.GetName(), nodeInfo.Node().GetName(), predicateKey, equivClassInfo.hash)
 	if !invalid {
 		return fit, reasons, nil

